reflect.go

```diff
@@ -108,7 +108,14 @@ type customSchemaImpl interface {
 	JSONSchema() *Schema
 }
 
+// Function to be run after the schema has been generated.
+// this will let you modify a schema afterwards
+type postSchemaImpl interface {
+	JSONSchemaPost(*Schema)
```

Comment:
I'm not totally convinced on the name...  "Post" just isn't obvious enough in my opinion.

Reply:
I think `JSONSchemaExtend` is best. If you're good with that I can rename it

Reply:
`JSONSchemaExtension` is perhaps more grammatically and semantically correct :-) (naming things 🤦 )

Reply:
I liked `JSONSchemaExtend` better, but your choice. Just let me know your final answer and I'll update the PR

Reply:
Cool, lets go with `JSONSchemaExtend`!

Reply:
Alright, renamed